sktime/utils/_testing/scenarios.py

```diff
@@ -0,0 +1,200 @@
+        names for keys need not equal names of methods
+
+    Methods
+    -------
+    run(obj, args=None, default_method_sequence=None)
+        Run a call(args) scenario on obj, and retrieve method outputs.
+    is_applicable(obj)
+        Check whether scenario is applicable to obj.
+    """
+
+    def __init__(
+        self, args=None, default_method_sequence=None, default_arg_sequence=None
+    ):
+        if default_method_sequence is not None:
+            self.default_method_sequence = _check_list_of_str(default_method_sequence)
+        if default_arg_sequence is not None:
+            self.default_arg_sequence = _check_list_of_str(default_arg_sequence)
+        if args is not None:
+            self.args = _check_dict_of_dict(args)
+        else:
+            if not hasattr(self, args):
+                raise RuntimeError(
+                    "args must either be given to __init__ or set in a child class"
+                )
+            _check_dict_of_dict(self.args)
+
+    def run(self, obj, method_sequence=None, arg_sequence=None, return_all=False):
+        """Run a call(args) scenario on obj, and retrieve method outputs.
+
+        Runs a sequence of commands
+            res_1 = obj.method_1(**args_1)
+            res_2 = obj.method_2(**args_2)
+            etc, where method_i is method_sequence[i],
+                and args_i is arg_sequence[i]
+        and returns results.
+
+        if method_i is __init__ (a constructor),
+        obj is changed to obj.__init__(**args_i) from the next line on
+
+        Parameters
+        ----------
+        obj : class or object with methods in method_sequence
+        method_sequence : list of str, default = arg_sequence if passed
+            if arg_sequence is also None, then default = self.default_method_sequence
+            sequence of method names to be run
+        arg_sequence : list of str, default = method_sequence if passed
+            if method_sequence is also None, then default = self.default_arg_sequence
+            sequence of keys for keyword argument dicts to be used
+            names for keys need not equal names of methods
+        return_all : bool, default = False
+            whether all or only the last result should be returned
+            if False, only the last result is returned
+            if True, list of deepcopies of intermediate results is returned
+
+        Returns
+        -------
+        results: output of the last method call, if return_all = False
+            list of deepcopies of all outputs, if return_all = True
+        """
+        # if both None, fill with defaults if exist
+        if method_sequence is None and arg_sequence is None:
+            method_sequence = getattr(self, "default_method_sequence", None)
+            arg_sequence = getattr(self, "default_arg_sequence", None)
+
+        # if only one is None, fill one with the other
+        # this will also raise an error if both are still None
```

Comment:
you have not implemented this error yet!
```suggestion
        # this will also raise an error if both are still None
```

Reply:
it will raise an error (because it executes line 100 which evaluates to `_check_list_of_str(None)`, but the message is not descriptive. Will update this.

Reply:
done, hope better?

Reply:
yup, looks good